src/historywork/BatchWork.cpp

```diff
@@ -65,4 +69,27 @@ BatchWork::notify(std::string const& child)
     mApp.getCatchupManager().logAndUpdateCatchupStatus(true);
     advance();
 }
+
+void
+BatchWork::handleNewWork(bool isFirst)
+{
+    auto newWork = yieldMoreWork();
+
+    if (!isFirst)
```

Comment:
not sure I understand the logic of `isFirst`: shouldn't that just be `if (mLastAssignedWork) {` ?